File: chap11/fig11-18.R
library(ggplot2)

d <- read.csv('input/data-eg4.csv')

d[d$Month == 8, ]$Y <- NA
d$Time <- 1:nrow(d)

p <- ggplot(data=d, aes(Time, Y)) +
  theme_bw(base_size=18) +
  geom_line(size=0.3) +
  labs(x='Time (Day)', y='Y')
ggsave(p, file='output/fig11-18.png', dpi=300, w=4, h=3)

File: chap11/fig11-18.py
import pandas
import numpy as np
from plotnine import ggplot, aes, theme_bw, geom_line, labs

d = pandas.read_csv('input/data-eg4.csv')
d.loc[d.Month == 8, 'Y'] = np.nan
d['Time'] = range(1, len(d)+1)

p = (ggplot(d, aes('Time', 'Y'))
    + theme_bw(base_size=18)
    + geom_line(size=0.3)
    + labs(x='Time (Day)', y='Y')
)
p.save(filename='output/fig11-18.py.png', dpi=300, width=4, height=3)
